src/Microsoft.TemplateEngine.Edge/Constraints/ConstraintsExtensions.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.TemplateEngine.Utils;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.TemplateEngine.Edge.Constraints
+{
+    internal static class ConstraintsExtensions
+    {
+        public static IEnumerable<string> ParseArrayOfConstraintStrings(this string? args)
```

Comment:
documentation might be useful here, especially knowing that the method can throw.